python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/test_asset_daemon.py

```diff
@@ -1,177 +1,76 @@
-        assert run_request.partition_key == expected_run_request.partition_key
 
-
-@pytest.fixture
-def instance():
-    with instance_for_test() as the_instance:
-        yield the_instance
-
-
-@pytest.mark.parametrize(
-    "scenario_item",
-    list(ASSET_RECONCILIATION_SCENARIOS.items()),
-    ids=list(ASSET_RECONCILIATION_SCENARIOS.keys()),
+from .test_scenarios import all_scenarios
+from .updated_scenarios.asset_daemon_scenario_states import (
+    two_assets_in_sequence,
+    two_partitions_def,
 )
-def test_reconcile_with_external_asset_graph(scenario_item, instance):
-    scenario_name, scenario = scenario_item
-    if not scenario.supports_with_external_asset_graph:
-        pytest.skip("Scenario does not support with_external_asset_graph")
-    run_requests, _, _ = scenario.do_sensor_scenario(
-        instance, scenario_name=scenario_name, with_external_asset_graph=True
-    )
 
-    assert len(run_requests) == len(scenario.expected_run_requests), run_requests
 
-    _assert_run_requests_match(scenario.expected_run_requests, run_requests)
-
-
-daemon_scenarios = {
-    **auto_materialize_policy_scenarios,
-    **multi_code_location_scenarios,
-    **auto_observe_scenarios,
-}
-
-
-@pytest.fixture
-def daemon_paused_instance():
+@contextmanager
+def get_daemon_instance(
+    paused: bool = False,
+    extra_overrides: Optional[Mapping[str, Any]] = None,
+) -> Generator[DagsterInstance, None, None]:
     with instance_for_test(
         overrides={
             "run_launcher": {
                 "module": "dagster._core.launcher.sync_in_memory_run_launcher",
                 "class": "SyncInMemoryRunLauncher",
             },
-            "auto_materialize": {
-                "max_tick_retries": 2,
-            },
+            **(extra_overrides or {}),
         }
-    ) as the_instance:
-        yield the_instance
-
-
-@pytest.fixture
-def daemon_not_paused_instance(daemon_paused_instance):
-    set_auto_materialize_paused(daemon_paused_instance, False)
-    return daemon_paused_instance
-
-
-def test_daemon_ticks(daemon_paused_instance):
-    instance = daemon_paused_instance
-    ticks = instance.get_ticks(
-        origin_id=FIXED_AUTO_MATERIALIZATION_ORIGIN_ID,
-        selector_id=FIXED_AUTO_MATERIALIZATION_SELECTOR_ID,
-    )
-    assert len(ticks) == 0
-
-    scenario = daemon_scenarios["auto_materialize_policy_lazy_freshness_missing"]
-    scenario.do_daemon_scenario(
-        instance, scenario_name="auto_materialize_policy_lazy_freshness_missing"
-    )
+    ) as instance:
+        set_auto_materialize_paused(instance, paused)
+        yield instance
 
-    ticks = instance.get_ticks(
-        origin_id=FIXED_AUTO_MATERIALIZATION_ORIGIN_ID,
-        selector_id=FIXED_AUTO_MATERIALIZATION_SELECTOR_ID,
-    )
 
-    # Daemon paused, so no ticks
-    assert len(ticks) == 0
+# take an arbitrary 10% of the scenarios to run end-to-end against the daemon
```

Comment:
its a little weird that this changes non-deterministically when you add a new scenario in the middle of the list. Did you consider hand-picking a certain number of scenarios instead?

Reply:
yeah I had a similar concern

Reply:
hand-picking is definitely reasonable, I think for now I'll just have it test over 100% of the scenarios, and if this step starts taking too long, we can always hand-pick later.